lib/stdlib/src/queue.erl

```diff
@@ -407,11 +407,10 @@ filter_r(Fun, [X|R0]) ->
 %%
 %% O(len(Q1))
 -spec filtermap(Fun, Q1) -> Q2 when
```

Comment:
I would write it
```
-spec filtermap(Fun, Q1) -> Q2 when
      Fun :: fun((Item) -> boolean() | {true, Item1}),
      Q1 :: queue(Item),
      Q2 :: queue(Item | Item1).
```

Reply:
Keep in mind that no item from `Q1` is preserved. They are all either filtered out (if the function returns `false`) or turned into whatever the function returns (if it returns `{true, …}`). That's why `Item` doesn't appear in `Q2`'s item type.

Reply:
an item from `Q1` can be preserved:

```
8> Q = queue:from_list([a, b, c, 1, 2, 3]).
{[3,2,1],[a,b,c]}
9> queue:filtermap(fun erlang:is_atom/1, Q).
{[c,b],[a]}
```

`Fun` returns `boolean()`. When it returns `true` the item is preserved.

Reply:
I also missed. Thank you for your help.